master/internal/rm/agentrm/resource_pool.go

```diff
@@ -731,38 +730,27 @@ func (rp *resourcePool) receiveJobQueueMsg(ctx *actor.Context) error {
 }
 
 func (rp *resourcePool) setGroupPriority(ctx *actor.Context, msg sproto.SetGroupPriority) error {
-	g := rp.getOrCreateGroup(ctx, msg.Handler)
+	g := rp.getOrCreateGroup(msg.JobID)
 	if (g.Priority != nil && *g.Priority == msg.Priority) ||
 		rp.config.Scheduler.Priority == nil {
 		return nil
 	}
-	ctx.Log().Infof("setting priority for group of %s to %d",
-		msg.Handler.Address().String(), msg.Priority)
+	ctx.Log().Infof("setting priority for group of %s to %d", msg.JobID, msg.Priority)
 	g.Priority = &msg.Priority
-	jobID, ok := rp.groupActorToID[msg.Handler]
-	if ok {
-		time, err := tasklist.GetJobSubmissionTime(rp.taskList, jobID)
-		if err != nil {
-			ctx.Log().Errorf("failed to get job submission time: %s", err)
-			return nil
-		}
-		rp.queuePositions[jobID] = tasklist.InitializeQueuePosition(time, false)
+	time, err := tasklist.GetJobSubmissionTime(rp.taskList, msg.JobID)
+	if err != nil {
+		err = errors.Wrapf(err, "failed to get job submission time for %s", msg.JobID)
+		ctx.Log().Error(err)
```

Comment:
nit handle errors once

Reply:
oddly this function does not return any errors, will leave that to another PR tho